tests/integration/trace/test_zipkin_trace.py

```diff
@@ -176,3 +178,17 @@ def hello(request):
 
     yield f
     assert count[0] == 1
+
+
+@pytest.mark.gen_test
+def test_zipkin_trace_sampling(trace_server):
+    run_times = 100000
+    sample_rate = 1.0 - random.random()
+    hook = ZipkinTraceHook(sample_rate=sample_rate)
+    count = 0
+    for _ in range(run_times):
+        if hook._lucky(_uniq_id()):
+            count += 1
+
+    assert 0.9 * run_times * sample_rate <= count
```

Comment:
My issue is with this specific test trying to measure randomness. I'm suggesting deleting this test in favor of something like,

    hook = ZipkinTraceHook(...)
    with mock(hook, '_lucky') as mock_lucky:
        mock_lucky.return_value = True
        request = Request()
        hook.before_send_request(request)
        assert ...

Usually we'd want to avoid something like that because of mocking `_lucky` but in this case there's no way for us to reasonably control the randomness without relying on more specific details of how `_lucky` is implemented.